File: 0318/SeoyeonHong/1018 체스판 다시 칠하기.py
import sys

input = sys.stdin.readline
N, M = map(int, input().split())
board = []
for _ in range(N):
    board.append(input())
min_count = N * M

def count_square(i, j):
    white, black = 0, 0
    color = 'W'
    for r in range(i, i+8):
        color = 'B' if color == 'W' else 'W'
        for c in range(j, j+8):
            if board[r][c] != color:
                white += 1
            else:
                black += 1
            if white >= min_count and black >= min_count:
                return min_count
            color = 'B' if color == 'W' else 'W'
    
    return min(white, black)

for i in range(N-7):
    for j in range(M-7):
        min_count = min(min_count, count_square(i, j))

print(min_count)

File: 0318/SoohyunKim/1018_체스판다시칠하기.js
const input = require('fs').readFileSync('/dev/stdin').toString().trim().split('\n');
const [N, M] = input.shift().split(' ').map(Number);
const board = input.map(line => line.split(''));
let min = Infinity;

const countColor = (x, y) => {
  let color = board[x][y]; // B or W
  let change = 0; // color 와 색이 같아야 하는지 달라야 하는지 구분하기 위한 변수
  let cnt = 0; // color로 시작
  let oppCnt = 0; // 반대 color로 시작
  
  for(let r = 0; r < 8; r++){
    if(r % 2 === 0) change = 0;
    else change = 1;

    for(let c = 0; c < 8; c++){
      if(change % 2 === 0){
        if(board[x+r][y+c] !== color) cnt++;
        if(board[x+r][y+c] === color) oppCnt++;
      } else {
        if(board[x+r][y+c] === color) cnt++;
        if(board[x+r][y+c] !== color) oppCnt++;
      }
      change++;
    }
  }

  return Math.min(cnt, oppCnt);
}

for(let i = 0; i <= N - 8; i++){
  for(let j = 0; j <= M - 8; j++){
    min = Math.min(min, countColor(i, j));
  }
}

console.log(min);